tasks.c

```diff
@@ -3188,7 +3188,13 @@ static void prvInitialiseNewTask( TaskFunction_t pxTaskCode,
                     /* The scheduler is not running, but the task that was pointed
                      * to by pxCurrentTCB has just been suspended and pxCurrentTCB
                      * must be adjusted to point to a different task. */
-                    if( listCURRENT_LIST_LENGTH( &xSuspendedTaskList ) == uxCurrentNumberOfTasks )
+
+                    /* Use temp variable as distinct sequence points for reading
+                     * volatile variables prior to a comparison, per the coding guidelines */
+                    UBaseType_t uxCurrentListLength = listCURRENT_LIST_LENGTH( &xSuspendedTaskList );
+                    UBaseType_t uxCurrentNumOfTasks = uxCurrentNumberOfTasks;
+
+                    if( uxCurrentListLength == uxCurrentNumOfTasks )
```

Comment:
Can we reduce one temp variable with the following example? Then only one volatile variable is used in comparison.
```suggestion
                    /* Use temp variable as distinct sequence points for reading
                     * volatile variables prior to a comparison, per the coding guidelines */
                    UBaseType_t uxCurrentListLength = listCURRENT_LIST_LENGTH( &xSuspendedTaskList );

                    if( uxCurrentListLength == uxCurrentNumberOfTasks )
```

Reply:
I do not have a strong preference towards either one, the reason I decided to assign both to temp variables was I thought it made the intent more clear.